packages/@aws-cdk/aws-lambda/lib/lambda-version.ts

```diff
@@ -81,11 +82,12 @@ export class Version extends QualifiedFunctionBase implements IVersion {
    * @param versionArn The version ARN to create this version from
    */
   public static fromVersionArn(scope: Construct, id: string, versionArn: string): IVersion {
-    const version = extractVersionFromArn(versionArn);
+    const version = extractQualifierFromArn(versionArn);
     const lambda = Function.fromFunctionArn(scope, `${id}Function`, versionArn);
 
     class Import extends QualifiedFunctionBase implements IVersion {
       public readonly version = version;
+      public readonly qualifier = version;
```

Comment:
`version` is an `@attribute`... not sure we can deprecate it, will it ever be removed?

Reply:
This will definitely happen with the next major version (v2). We don't have any plans to remove these before that, unless something comes along to expedite it.

Everything flagged as deprecated will be considered when we start looking at this.

Reply:
Yes, of course but isn't the fact that it's an attribute and that all CF attributes must be exposed as-is conflicting here? (marked as `@attribute`).

Reply:
There shouldn't be a problem with actually deprecating methods with the `@attribute`. 

If not already, the awslint should support specifying an optional name parameter to the `@attribute` tag so there's the flexibility to have CDK's members named differently from CloudFormation's return types, as so `@attribute('name')` or `@attribute 'name'`.

I tried to take a stab at this myself, but it's a bit more intrusive than expected and the value to making the change doesn't yet pay off. There's a bunch of other places like Alias that also needs to have similar changes and I'm going to refrain from going all the way.

For now, let's leave this as is. I've reduced the visibility of the `qualifier` member to `protected`, so consumers of the module use what's already there without getting confused.